test/TesterInternal/GeoClusterTests/LogViewProviderTestBattery.cs

```diff
@@ -0,0 +1,490 @@
+﻿using Orleans;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.TestingHost;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnitTests.GrainInterfaces;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Tests.GeoClusterTests
+{
+    /// <summary>
+    ///  A fixture that provides a collection of semantic tests for log view providers
+    ///  (concurrent reading and updating, update propagation, conflict resolution)
+    ///  on a multicluster with the desired number of clusters
+    /// </summary>
+    public class LogViewProviderTestFixture : TestingClusterHost
+    {
+
+        #region client wrappers
+
+        public class ClientWrapper : Tests.GeoClusterTests.TestingClusterHost.ClientWrapperBase
+        {
+            public ClientWrapper(string name, int gatewayport, string clusterId, Action<ClientConfiguration> customizer)
+               : base(name, gatewayport, clusterId, customizer)
+            {
+                systemManagement = GrainClient.GrainFactory.GetGrain<IManagementGrain>(RuntimeInterfaceConstants.SYSTEM_MANAGEMENT_ID);
+            }
+
+            public string GetGrainRef(string grainclass, int i)
+            {
+                return GrainClient.GrainFactory.GetGrain<ISimpleLogViewGrain>(i, grainclass).ToString();
+            }
+
+            public void SetALocal(string grainclass, int i, int a)
+            {
+                var grainRef = GrainClient.GrainFactory.GetGrain<ISimpleLogViewGrain>(i, grainclass);
+                grainRef.SetALocal(a).Wait();
```

Comment:
Yes, that makes sense.

The functions are synchronous because they are called from a different AppDomain (the class derives from MarshalByRefObject).  Don't know if there is another way to do it.